src/cpp/src/utils.cpp

```diff
@@ -266,6 +268,21 @@ ov::Core singleton_core() {
     return core;
 }
 
+bool is_tokenized_history_same(const ov::Tensor& encoded_history, const std::vector<int64_t> tokenized_history) {
+    size_t i = 0;
+    // encoded_history should be eq or less then tokenized_history, tokenized_history also include eos token
+    if (encoded_history.get_size() > tokenized_history.size())
+        return false;
+    auto encoded_history_data = encoded_history.data<int64_t>();
+    while(i < encoded_history.get_size()) {
+        if (encoded_history_data[i] != tokenized_history[i])
+            break;
+        i++;
+    }
+
+    return i == tokenized_history.size() - 1 || i == tokenized_history.size() - 2;
```

Comment:
Why is `i == tokenized_history.size() - 2` condition treated as same history? The last tokens are different in that case